library/init/datatypes.lean

```diff
@@ -394,23 +394,23 @@ definition default_has_sizeof_eq (A : Type) (a : A) : @sizeof A (default_has_siz
 rfl
 
 attribute [instance]
-definition nat_has_sizeof : has_sizeof nat :=
+definition nat.has_sizeof : has_sizeof nat :=
```

Comment:
> I am relying on finding these declarations in the C++ by appending has_sizeof to the name of the inductive type in question

`has_sizeof` is a type class. So, we can use type class resolution to find the `has_sizeof` instance for a type `T`.

> I think we should handle this the same way we handle .cases_on, and protect to avoid the overloads.

`.cases_on` is different. First, there is no top level `cases_on` declaration. Second, `cases_on` is not a type class.

In general, the names of type class instances do not matter. For example, we don't need any naming convention for handling `has_add` instances in the automation.

If you need a naming convention, please use something like `.has_sizeof_inst`. 
Anyway, it looks weird that the code only works if the instance for `nat` is called `nat.has_size_of`.